emcc.py

```diff
@@ -1313,18 +1326,19 @@ def check(input_file):
       if any(s.startswith('MEM_INIT_METHOD=') for s in settings_changes):
         exit_with_error('MEM_INIT_METHOD is not supported in wasm. Memory will be embedded in the wasm binary if threads are not used, and included in a separate file if threads are used.')
       options.memory_init_file = True
-      if shared.Settings.BINARYEN_ASYNC_COMPILATION == 1:
-        # async compilation requires a swappable module - we swap it in when it's ready
+      if shared.Settings.BINARYEN_ASYNC_COMPILATION == 1 and not shared.Settings.MINIMAL_RUNTIME:
+        # async compilation requires a swappable module - we swap it in when it's ready (MINIMAL_RUNTIME does not need to utilize the swappable module concept)
```

Comment:
The way MINIMAL_RUNTIME JS file is written, the Wasm is always asynchronously compiled, and the SWAPPABLE_ASM_MODULE mechanism does need to be used